sentry-rails/lib/sentry/rails/breadcrumb/active_support_logger.rb

```diff
@@ -2,72 +2,12 @@ module Sentry
-          # action_dispatch
-          "process_middleware.action_dispatch" => %i[middleware],
-          # action_view
-          "render_template.action_view" => %i[identifier layout],
-          "render_partial.action_view" => %i[identifier],
-          "render_collection.action_view" => %i[identifier count cache_hits],
-          "render_layout.action_view" => %i[identifier],
-          # active_record
-          "sql.active_record" => %i[sql name statement_name cached],
-          "instantiation.active_record" => %i[record_count class_name],
-          # action_mailer
-          # not including to, from, or subject..etc. because of PII concern
-          "deliver.action_mailer" => %i[mailer date perform_deliveries],
-          "process.action_mailer" => %i[mailer action params],
-          # active_support
-          "cache_read.active_support" => %i[key store hit],
-          "cache_generate.active_support" => %i[key store],
-          "cache_fetch_hit.active_support" => %i[key store],
-          "cache_write.active_support" => %i[key store],
-          "cache_delete.active_support" => %i[key store],
-          "cache_exist?.active_support" => %i[key store],
-          # active_job
-          "enqueue_at.active_job" => %i[],
-          "enqueue.active_job" => %i[],
-          "enqueue_retry.active_job" => %i[],
-          "perform_start.active_job" => %i[],
-          "perform.active_job" => %i[],
-          "retry_stopped.active_job" => %i[],
-          "discard.active_job" => %i[],
-          # action_cable
-          "perform_action.action_cable" => %i[channel_class action],
-          "transmit.action_cable" => %i[channel_class],
-          "transmit_subscription_confirmation.action_cable" => %i[channel_class],
-          "transmit_subscription_rejection.action_cable" => %i[channel_class],
-          "broadcast.action_cable" => %i[broadcasting],
-          # active_storage
-          "service_upload.active_storage" => %i[service key checksum],
-          "service_streaming_download.active_storage" => %i[service key],
-          "service_download_chunk.active_storage" => %i[service key],
-          "service_download.active_storage" => %i[service key],
-          "service_delete.active_storage" => %i[service key],
-          "service_delete_prefixed.active_storage" => %i[service prefix],
-          "service_exist.active_storage" => %i[service key exist],
-          "service_url.active_storage" => %i[service key url],
-          "service_update_metadata.active_storage" => %i[service key],
-          "preview.active_storage" => %i[key],
-          "analyze.active_storage" => %i[analyzer],
-        }.freeze
-
         class << self
           def add(name, started, _finished, _unique_id, data)
             # skip Rails' internal events
             return if name.start_with?("!")
 
-            allowed_keys = ALLOWED_LIST[name]
+            allowed_keys = Sentry.configuration.rails.active_support_logger_subscription_items[name]
```

Comment:
can you pass in the configuration list from `inject` like this? there are some multi-threaded edge cases with accessing `Sentry.configuration` that this avoids.

```patch
➜  sentry-rails git:(igordepolli/master) ✗ git --no-pager diff
diff --git a/sentry-rails/lib/sentry/rails/breadcrumb/active_support_logger.rb b/sentry-rails/lib/sentry/rails/breadcrumb/active_support_logger.rb
index 3af825d2..9dfdd1d2 100644
--- a/sentry-rails/lib/sentry/rails/breadcrumb/active_support_logger.rb
+++ b/sentry-rails/lib/sentry/rails/breadcrumb/active_support_logger.rb
@@ -7,10 +7,8 @@ module Sentry
             # skip Rails' internal events
             return if name.start_with?("!")

-            allowed_keys = Sentry.configuration.rails.active_support_logger_subscription_items[name]
-
             if data.is_a?(Hash)
-              data = data.slice(*allowed_keys)
+              data = data.slice(*@allowed_keys[name])
             end

             crumb = Sentry::Breadcrumb.new(
@@ -21,7 +19,9 @@ module Sentry
             Sentry.add_breadcrumb(crumb)
           end

-          def inject
+          def inject(allowed_keys)
+            @allowed_keys = allowed_keys
+
             @subscriber = ::ActiveSupport::Notifications.subscribe(/.*/) do |name, started, finished, unique_id, data|
               # we only record events that has a started timestamp
               if started.is_a?(Time)
diff --git a/sentry-rails/lib/sentry/rails/railtie.rb b/sentry-rails/lib/sentry/rails/railtie.rb
index 4e04d0e3..f77a2b7e 100644
--- a/sentry-rails/lib/sentry/rails/railtie.rb
+++ b/sentry-rails/lib/sentry/rails/railtie.rb
@@ -89,7 +89,7 @@ module Sentry
     def inject_breadcrumbs_logger
       if Sentry.configuration.breadcrumbs_logger.include?(:active_support_logger)
         require 'sentry/rails/breadcrumb/active_support_logger'
-        Sentry::Rails::Breadcrumb::ActiveSupportLogger.inject
+        Sentry::Rails::Breadcrumb::ActiveSupportLogger.inject(Sentry.configuration.rails.active_support_logger_subscription_items)
       end

       if Sentry.configuration.breadcrumbs_logger.include?(:monotonic_active_support_logger)
```